internal/migration/migration.go

```diff
@@ -0,0 +1,478 @@
+package migration
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	json "github.com/nikkolasg/hexjson"
+	"go.etcd.io/bbolt"
+
+	"github.com/drand/drand/chain"
+	"github.com/drand/drand/chain/postgresdb/database"
+	"github.com/drand/drand/chain/postgresdb/pgdb"
+	"github.com/drand/drand/chain/postgresdb/schema"
+	"github.com/drand/drand/log"
+)
+
+type (
+	beacon struct {
+		PreviousSig []byte `json:",omitempty"`
+		Round       uint64
+		Signature   []byte
+	}
+
+	migrator struct {
+		ctx       context.Context
+		ctxCancel context.CancelFunc
+
+		startedAt time.Time
+		logger    log.Logger
+
+		pgDSN            string
+		bufferSize       int
+		beaconName       string
+		sourceBeaconPath string
+
+		// existingDB   *bbolt.DB
+		existingRows int
+
+		distChan    chan beacon
+		errChan     chan error
+		destination chain.StorageType
+	}
+)
+
+const ownerOnly = 0600
+
+const DefaultBufferSize = 10_000
+
+var (
+	// ErrMigrationNotNeeded ...
+	ErrMigrationNotNeeded = fmt.Errorf("migration not needed")
```

Comment:
Since you put this in a var group, we could have a comment above the var.

Now that being said, you are defining something other than an error variable. So let's remove the group, attach var to each and apply a better comment to the error variable.